mlflow/xgboost.py

```diff
@@ -698,3 +492,262 @@ def infer_model_signature(input_example):
 
     safe_patch(FLAVOR_NAME, xgboost, "train", train, manage_run=True)
     safe_patch(FLAVOR_NAME, xgboost.DMatrix, "__init__", __init__)
+
+    # enable xgboost scikit-learn estimators autologging
+    import mlflow.sklearn
+
+    xgb_sklearn_autolog = autologging_integration(FLAVOR_NAME_SKLEARN)(mlflow.sklearn._autolog)
+    xgb_sklearn_autolog(
+        flavor_name=FLAVOR_NAME_SKLEARN,
+        log_input_examples=log_input_examples,
+        log_model_signatures=log_model_signatures,
+        log_models=log_models,
+        disable=disable,
+        exclusive=exclusive,
+        disable_for_unsupported_versions=disable_for_unsupported_versions,
+        silent=silent,
+        max_tuning_runs=max_tuning_runs,
+        log_post_training_metrics=True,
+        xgboost_estimator=True,
+    )
+
+
+def _mlflow_xgboost_logging(
+    importance_types, autologging_client, logger, original, sklearn_estimator, *args, **kwargs,
+):
```

Comment:
Re-organize early stopping call backs and feature importance plot. This function is re-used in `mlflow.sklearn` for logging XGBoost sklearn estimators.